### ISSUE TITLE
Update sum with push and pop

### ISSUE THREAD

**Issue by hajduakos** [SRI-CSL/solidity#52]
Sum is not updated when pushing/popping arrays. The following example should be reported correct:
```solidity
/// @notice invariant __verifier_sum_uint(items) == total
contract C {
    uint total;
    uint[] items;

    function add(uint v) public {
        items.push(v);
        total += v;
    }

    function remove() public {
        total -= items[items.length-1];
        items.pop();
    }
}
```

**Comment by hajduakos** [comment]
Fixed in ffdb0c7, but we should only close the issue when it is merged

**Comment by hajduakos** [comment]
Merged, closing issue
